.github/workflows/pull-request.yml

```diff
@@ -1,6 +1,10 @@
 name: PR build
 
-on: [push, pull_request]
+on:
+  pull_request:
+  push:
+    branches-ignore:
+      - master
```

Comment:
@ThisaruGuruge will this try to publish to codecov from other branches?? We should push only from master.

Reply:
Yes. This will publish when there's a push on the other branches. I think we can omit this by adding a condition to CodeCov publish step.

Reply:
Changed `pull-request.yml` to only run codecov action on PR